cmd/kube-proxy/app/server.go

```diff
@@ -146,6 +152,13 @@ func (s *ProxyServer) Run(_ []string) error {
 		}, 5*time.Second)
 	}
 
+	go func() {
+		err := proxy.ServeDNS(dnsHandler)
```

Comment:
This is typically in a go util.Forever loop (if ServeDNS doesn't spawn its own goroutines).  If it does, this is fine.